src/main/java/jenkins/plugins/gerrit/GerritURI.java

```diff
@@ -0,0 +1,101 @@
+package jenkins.plugins.gerrit;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.jgit.transport.URIish;
+
+/**
+ * A GerritURI encapsulates a Gerrit remote URI and is able to extract certain properties like the project name and URI
+ * prefix.
+ */
+public class GerritURI {
+
+  private static enum Scheme {
+    HTTP,
+    HTTPS,
+    SSH
+  }
+
+  /**
+   * Pattern which matches the URI prefix and the project name of a Gerrit HTTP URI.
+   */
+  private static final Pattern GERRIT_HTTP_URI_PATTERN = Pattern.compile("(.*?)/a/(.*)");
+
+  private final URIish remoteURI;
+
+  /**
+   * Create a new Gerrit URI from the given remote {@link URIish URI}.
+   *
+   * @param remoteURI the remote URI, e.g. https://host/a/project
+   */
+  public GerritURI(URIish remoteURI) {
+    this.remoteURI = remoteURI;
+  }
+
+  /**
+   * Get the full Gerrit remote URI.
+   *
+   * @return the remote URI
+   */
+  public URIish getRemoteURI() {
+    return remoteURI;
+  }
+
+  /**
+   * Get the Gerrit project name from the remote URI.
+   *
+   * @return the Gerrit project name
+   */
+  public String getProject() {
+    Scheme scheme = getScheme(remoteURI.getScheme());
+    switch (scheme) {
+      case HTTP:
+      case HTTPS:
+        Matcher matcher = GERRIT_HTTP_URI_PATTERN.matcher(remoteURI.getRawPath());
```

Comment:
Why not including the ```(http)s?://``` in ```GERRIT_HTTP_URI_PATTERN``` directly? We won't have anymore the need to define the enum and the associated switch :-)

Reply:
This was part of an attempt to add SSH support, which is a bit more complex than I expected. (And the path doesn't contain the scheme, it only contains the path part of the URI).

I can try to simplify this change, but i'm not sure how to address the issue mentioned by @grotrek.

Reply:
I think the best way is adding option field for API URI. Because Gerrit may be not only behind a reverse proxy but either on a non-standard port. @lucamilanesio, what are you thinking about it?

Reply:
@grotrek @auke- apologies for the delay of picking this up. I believe this is good for now. Once we'll start supporting SSH, then we will need to "guess" the API URL or, alternatively, just issue the review commands over SSH instead of HTTP. But that is going to be discussed in a separate PR when we'll support SSH.